utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,172 @@
+    return test_hists
+
+def move(case, suffix):
+    """
+    Change the suffix for the most recent batch of hist files
+    """
+    rundir   = case.get_value("RUNDIR")
+    testcase = case.get_value("CASE")
+
+    # Loop over models
+    comments = "Moving hist files to suffix '%s'\n" % suffix
+    num_moved = 0
+    for model in _iter_model_file_substrs():
+        comments += "  Moving hist files for model '%s'\n" % model
+        test_hists = _get_all_hist_files(testcase, model, from_dir=rundir)
+        num_moved += len(test_hists)
+        for test_hist in test_hists:
+            new_file = "%s.%s" % (test_hist, suffix)
+            if os.path.exists(new_file):
+                os.remove(new_file)
+
+            comments += "    Copying '%s' to '%s'\n" % (test_hist, new_file)
+            shutil.copy(test_hist, new_file)
+
+    expect(num_moved > 0, "move failed: no hist files found in rundir '%s'" % rundir)
+
+    return comments
+
+def _compare_hists(case, suffix1="", suffix2="", from_dir1=os.getcwd(), from_dir2=os.getcwd()):
+
+    testcase = case.get_value("CASE")
+    all_success = True
+    num_compared = 0
+    comments = "Comparing hists for case '%s' dir1='%s', suffix1='%s',  dir2='%s' suffix2='%s'\n" % \
+        (testcase, from_dir1, suffix1, from_dir2, suffix2)
+
+    for model in _iter_model_file_substrs():
+        comments += "  comparing model '%s'\n" % model
+        hists1 = _get_all_hist_files(testcase, model, suffix1, from_dir1)
+        hists2 = _get_all_hist_files(testcase, model, suffix2, from_dir2)
+        if len(hists1) != len(hists2):
+            comments += "    num hists does not match %d != %d\n" % (len(hists1), len(hists2))
+            all_success = False
+            continue
+
+        num_compared += len(hists1)
+        for hist1, hist2 in zip(hists1, hists2):
+            success, cprnc_comments = cprnc(hist1, hist2, case, from_dir1)
+            if success:
+                comments += "    %s matched %s\n" % (hist1, hist2)
+            else:
+                comments += "    %s did NOT match %s\n" % (hist1, hist2)
+                comments += cprnc_comments + "\n"
+                all_success = False
+
+    expect(num_compared > 0, "Did not compare any hist files for suffix1='%s' suffix2='%s', dir1='%s', dir2='%s'\nComments=%s" %
+           (suffix1, suffix2, from_dir1, from_dir2, comments))
+    return all_success, comments
+
+def compare_test(case, suffix1, suffix2):
+    """
+    Compares two component history files in the testcase directory
+
+    returns (SUCCESS, comments)
+    """
+    rundir   = case.get_value("RUNDIR")
+
+    return _compare_hists(case, suffix1, suffix2, rundir, rundir)
+
+def cprnc(file1, file2, case, rundir):
+    cprnc_exe = case.get_value("CCSM_CPRNC")
+    basename = os.path.basename(file1)
+    stat, out, _ = run_cmd("%s %s %s 2>&1 | tee %s/%s.cprnc.out" % (cprnc_exe, file1, file2, rundir, basename))
+    return (stat == 0 and "IDENTICAL" in out, out)
```

Comment:
This is not a new issue (it looks like this is just replicating the old behavior), so it doesn't necessarily need to be addressed in this PR, but if it isn't, I'd like to open a separate issue to have this fixed: It looks like the comparison passes as long as the string "IDENTICAL" is found _anywhere_ in the cprnc output. In the event that there is a history field with IDENTICAL in its name, the comparison would always look successful. It would be more robust to check against the exact string - i.e., looking for "diff_test: the two files seem to be IDENTICAL".